src/test/fuzz/deserialize.cpp

```diff
@@ -35,6 +39,15 @@ void initialize()
 {
     // Fuzzers using pubkey must hold an ECCVerifyHandle.
     static const auto verify_handle = MakeUnique<ECCVerifyHandle>();
+
+    // Enable logging. This avoids accumulating allocated memory in a global
+    // state (via m_msgs_before_open buffering) if LogPrintf is called when
+    // fuzzing. Log messages are written to standard output.
+    SelectParams(CBaseChainParams::REGTEST);
+    InitLogging();
```

Comment:
Pretty sure this will log to (and maybe corrupt) the users main data directory

Reply:
The regtest `debug.log` file you mean?

```
$ strace -f src/test/fuzz/messageheader_deserialize 2>&1 | grep -E '^open.*regtest'
openat(AT_FDCWD, "/…/.bitcoin/regtest/debug.log", O_WRONLY|O_CREAT|O_APPEND, 0666) = 3
```

That's not very nice and we shouldn't do that. The reason was that the `m_print_to_file` is effectively defaulting to `true` due to the initialization logic. It is easy to miss that since it is initialised to `false` first:

```
$ git grep 'bool m_print_to_file'
src/logging.h:        bool m_print_to_file = false;
```

Good catch! Thanks!

Now logging to `/dev/null` instead via `::gArgs.ForceSetArg("-debuglogfile", "/dev/null");`:

```
$ strace -f src/test/fuzz/messageheader_deserialize 2>&1
openat(AT_FDCWD, "/dev/null", O_WRONLY|O_CREAT|O_APPEND, 0666) = 3
```

Let me know if you think it should be solved in some other way. I'm open to suggestions :)

Reply:
In the gui tests we use this hack:

https://github.com/bitcoin/bitcoin/blob/2ed74a43a05a47129d56117deeb489addbcaf05f/src/qt/test/test_main.cpp#L45-L53

Reply:
If you decide to use this hack, it might make sense to move it to `src/test/util/setup_common.h` so that it can be more easily re-used.

Reply:
I think I prefer the more explicit setup method: that way all necessary setup preconditions for the fuzzer are clearly stated in the fuzzing harness.

Now setting both `-datadir` (dummy) and `-debuglogfile` (`/dev/null`) to make sure there is no possibility of interfering with the main regtest location.

Please re-review :)